homeassistant/components/acaia/sensor.py

```diff
@@ -0,0 +1,141 @@
+"""Sensor platform for Acaia."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import cast
+
+from aioacaia.acaiascale import AcaiaDeviceState, AcaiaScale
+from aioacaia.const import UnitMass as AcaiaUnitOfMass
+
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorExtraStoredData,
+    SensorStateClass,
+)
+from homeassistant.const import PERCENTAGE, UnitOfMass
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .coordinator import AcaiaConfigEntry, AcaiaCoordinator
+from .entity import AcaiaEntity
+
+
+@dataclass(kw_only=True, frozen=True)
+class AcaiaSensorEntityDescription(SensorEntityDescription):
+    """Description for Acaia sensor entities."""
+
+    unit_fn: Callable[[AcaiaDeviceState], str] | None = None
+    value_fn: Callable[[AcaiaScale], StateType]
+
+
+SENSORS: tuple[AcaiaSensorEntityDescription, ...] = (
+    AcaiaSensorEntityDescription(
+        key="weight",
+        translation_key="weight",
+        device_class=SensorDeviceClass.WEIGHT,
+        native_unit_of_measurement=UnitOfMass.GRAMS,
+        state_class=SensorStateClass.MEASUREMENT,
+        unit_fn=lambda data: (
+            UnitOfMass.OUNCES
+            if data.units == AcaiaUnitOfMass.OUNCES
+            else UnitOfMass.GRAMS
+        ),
```

Comment:
does it change on runtime?

Reply:
yes. Because I don't know what is set on the scale. And theoretically the user could change it